### DISCUSSION TITLE
Remove embedded thumbnail but retain colour profile?

### DISCUSSION THREAD

**Discussion by Trailboy** [ImageMagick/ImageMagick#4659] [General] [answered]
Hi all.  

Is there a command that can strip the embedded thumbnail from an image but retain the colour profile? 

-strip 

strips the thumbnail so that my tiled output thumbnails reflects correctly in windows previews, but also strips the embedded colour profile.   The attached image below shows the problem I currently have: without -strip some of my tiled images retain the original thumbnail preview in Windows, while others don't, which is causing confusion.  My IM version is also shown.

Thanks so much if you can help me with this problem.

![Capture](https://user-images.githubusercontent.com/64076927/147971810-1aab2b39-9fce-4429-9b92-d3a7e35d78e4.JPG)

**Comment by snibgo** [comment, marked as answer]
You can save the embedded ICC profile to a file, then "-strip", then re-assign the profile from the file:
```
magick in.jpg +write temp.icc -strip -profile temp.icc in.jpg
```

**Reply by Trailboy** [reply]
Thank you yet again, that works perfectly.
